experimental/PartitionedPermutations/src/EnumeratePartitionedPermutations.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    _enumerate_all_partitions(n::Int)
+
+Return and efficiently calculate vector of all partitions of length `n`.
+This function is a helper function for `_enumerate_partitioned_perm`.
+
+# Examples
+```jldoctest
+julia> length(_enumerate_all_partitions(6))
+203
+```
+"""
+function _enumerate_all_partitions(n::Int)
+
+    partitions = [[1]]
+    index_to_cut = 0
+    count = false
+    
+    for i in partitions
+        blocks = maximum(i)
+        length(i) == n && continue
+        count = (length(i) == n - 1)
+        for ii in 1:(blocks+1)
+            push!(partitions, vcat(i, ii))
+            if count
+                index_to_cut += 1
+            end
+        end
+    end
+    return last(partitions, index_to_cut == 0 ? 1 : index_to_cut)
+end
+
+"""
+    enumerate_partitioned_perm(n::Int)
+
+Return and calculate all `PartitionedPermutation` objects of length `n`
+
+# Examples
+```jldoctest
+julia> length(enumerate_partitioned_perm(6))
+4051
+```
+"""
+function enumerate_partitioned_perm(n::Int)
+    
+    partitioned_permutations = []
+
+    # Iterate over all permutations
+    for p in Generic.elements!(Generic.SymmetricGroup(n))
```

Comment:
The `SymmetricGroup` from AbstractAlgebra should not be used in real usage code. Please use `symmetric_group` (backed by GAP) instead.
(@fingolfin please correct me if I am wrong)

Reply:
It seems that we cannot use symmetric_group instead of Generic.SymmetricGroup. Indeed, the former yields an object of type `PermGroup` which is not accepted by the function `Generic.elements!`.

Reply:
You can just do `for p in symmetric_group(n)`.

Reply:
Well, then I get a problem later because elements of `symmetric_group(n)` are of type `PermGroupElem` and not of type `Perm`. Is it generally recommended to use the types `PermGroup` and `PermGroupElem` instead of `Generic.SymmetricGroup` and `Perm`, respectively?

Reply:
Yes, the `AbstractAlgebra.Generic.SymmetricGroup` is only a toy-like implementation that is not integrated into other parts of OSCAR. Thus, they should not be used outside of AbstractAlgebra.jl

Reply:
If functionality for `PermGroup` or `PermGroupElem` should be missing, let us know and somebody can add it.

Reply:
Here is a functionality that seems to be missing: We need a function like `cycles(g::Perm)` that returns a list of the cycles of the permutation `g`. Apparently, one can only obtain the cycle structure of a `PermGroupElem` with `cycle_structure(g::PermGroupElem)` but not the cycles explicitely.

Reply:
@fingolfin @ThomasBreuer could we add `cycles` for `PermGroupElem`?
Easiest would be to copy the code from `cycles` and call it on `Vector(g)` I guess.

Reply:
Thank you very much! We have now replaced every mention of Perm with PermGrouElem.